### PR TITLE
add atcoder modint

### LINKED ISSUE #3524: Atcoder ModInt
is used in `/gold/combo` but not in `/gold/modular`. it should probably be used in both

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

closes #3524 (or at least i hope it does)

### CODE REVIEW

File: content/4_Gold/Modular.mdx [hunk 1]
```diff
@@ -324,9 +329,21 @@ We can also use the extended Euclidean algorithm. See the module in the
...
+	</Resource>
+	<Resource
+		source="AtCoder"
+		title="ModInt Documentation"
+		url="https://atcoder.github.io/ac-library/document_en/modint.html"
+	/>
 </Resources>
 
-Using BenQ's template, both of these do the same thing:
+Here's an example using BenQ's template:
```
Comment (bqi343) [inline content/4_Gold/Modular.mdx:346]:
```suggestion
Here's an example using Benq's template:
```

File: content/4_Gold/Modular.mdx [hunk 2]
```diff
@@ -309,8 +309,13 @@ We can also use the extended Euclidean algorithm. See the module in the
 
 </Optional>
 
+<CPPOnly>
+
 ## Templates
 
+Here's some templates that implement numerical types that automatically
```
Comment (bqi343) [inline content/4_Gold/Modular.mdx:316]:
```suggestion
Here are some templates that implement integer types that automatically
```

File: content/4_Gold/Modular.mdx [hunk 3]
```diff
@@ -367,12 +384,30 @@ int main() {
...
+#include <bits/stdc++.h>
+using namespace std;
+
+#include <atcoder/modint>
+using mint = atcoder::modint;
+
+int main() {
+	mint a = 1e8, b = 1e8, c = 1e8;
+	mint res = a * b * c;
+	cout << res.val() << endl;  // Same result
```
Comment (bqi343) [inline content/4_Gold/Modular.mdx:405]: this code doesn't compile

Comment (SansPapyrus683) [reply content/4_Gold/Modular.mdx:405]: how should i fix it

Comment (bqi343) [reply content/4_Gold/Modular.mdx:405]: try running it locally and see what errors you get? we should only add this code if we expect users to actually run it. 

or we can not add this if we think the module is fine as is. I suggested adding it just to make it consistent w/ the other module.

**Review by bqi343 — APPROVED** [review]

**Pull request merged**